controllers/duty.ts

```diff
@@ -0,0 +1,111 @@
+import express, { Request, Response } from 'express';
+
+import {
+  createDuty,
+  findDutyByID,
+  findDutiesByQuery,
+  deleteDuty,
+  updateDuty,
+} from '../services/duty.service';
+import validateRequest from './validation';
+import { TypedRequestBody } from './types';
+import { ClientError } from '../error_handling/client_errors';
+import { Duty, DutyQuery, DutyUpdate, Id } from '../schemas/duty.zschema';
+import {
+  DeleteDutySchema,
+  GetDutyByIdSchema,
+  GetDutyByQuerySchema,
+  PatchDutySchema,
+  PostDutySchema,
+} from '../schemas/duty.schema';
+
+const router = express.Router();
+
+router.post(
+  '/',
+  validateRequest(PostDutySchema),
+  async (req: TypedRequestBody<Duty>, res: Response) => {
```

Comment:
Is there a way to name your DTO (data transfer objects) and types the way that you did with soldiers? . `PostDutyDto` instead of `Duty`